src/main/java/services/RestoreFactory.java

```diff
@@ -363,6 +373,9 @@ public String getRestoreUrl(boolean overwriteCache) {
             builder.append("&since=").append(syncToken);
         }
         */
+
+        builder.append("&device_id=" + getSyncDeviceId());
```

Comment:
The `RestTemplate` class handles URL Encoding for us automatically

Agreed Re: UriComponents, we use that for the claim and query screens but not here.